dds/DCPS/NetworkAddress.cpp

```diff
@@ -0,0 +1,263 @@
+#endif
+  }
+}
+
+NetworkAddress& NetworkAddress::operator=(const NetworkAddress& rhs)
+{
+  if (&rhs != this) {
+    memcpy(&inet_addr_, &rhs.inet_addr_, sizeof (inet_addr_));
+  }
+  return *this;
+}
+
+NetworkAddress& NetworkAddress::operator=(const ACE_INET_Addr& rhs)
+{
+  return *this = NetworkAddress(rhs);
+}
+
+bool NetworkAddress::operator==(const NetworkAddress& rhs) const
+{
+  return memcmp(&inet_addr_, &rhs.inet_addr_, sizeof (inet_addr_)) == 0;
+}
+
+bool NetworkAddress::operator!=(const NetworkAddress& rhs) const
+{
+  return memcmp(&inet_addr_, &rhs.inet_addr_, sizeof (inet_addr_)) != 0;
+}
+
+bool NetworkAddress::operator<(const NetworkAddress& rhs) const
+{
+  return memcmp(&inet_addr_, &rhs.inet_addr_, sizeof (inet_addr_)) < 0;
+}
+
+bool NetworkAddress::addr_bytes_equal(const NetworkAddress& rhs) const
+{
+  if (inet_addr_.in4_.sin_family == AF_INET) {
+    return memcmp(&inet_addr_.in4_.sin_addr, &rhs.inet_addr_.in4_.sin_addr, sizeof (inet_addr_.in4_.sin_addr)) == 0;
+#if defined (ACE_HAS_IPV6)
+  } else if (inet_addr_.in6_.sin6_family == AF_INET6) {
+    return memcmp(&inet_addr_.in6_.sin6_addr, &rhs.inet_addr_.in6_.sin6_addr, sizeof (inet_addr_.in6_.sin6_addr)) == 0;
+#endif
+  }
+  return 0;
+}
+
+ACE_INET_Addr NetworkAddress::to_addr() const
+{
+  ACE_INET_Addr result;
+  result.set_addr(const_cast<ip46*>(&inet_addr_), sizeof (inet_addr_));
+  return result;
+}
+
+void NetworkAddress::to_addr(ACE_INET_Addr& addr) const
+{
+  addr.set_addr(const_cast<ip46*>(&inet_addr_), sizeof (inet_addr_));
+}
+
+int16_t NetworkAddress::get_type() const
+{
+  return static_cast<int16_t>(inet_addr_.in4_.sin_family);
+}
+
+void NetworkAddress::set_type(int16_t type)
+{
+  inet_addr_.in4_.sin_family = type;
+}
+
+uint16_t NetworkAddress::get_port_number() const
+{
+  if (inet_addr_.in4_.sin_family == AF_INET) {
+    return ACE_NTOHS(inet_addr_.in4_.sin_port);
+#if defined (ACE_HAS_IPV6)
+  } else if (inet_addr_.in6_.sin6_family == AF_INET6) {
+    return ACE_NTOHS(inet_addr_.in6_.sin6_port);
+#endif
+  }
+  return 0;
+}
+
+void NetworkAddress::set_port_number(uint16_t port)
+{
+  if (inet_addr_.in4_.sin_family == AF_INET) {
+    inet_addr_.in4_.sin_port = ACE_HTONS(port);
+#if defined (ACE_HAS_IPV6)
+  } else if (inet_addr_.in6_.sin6_family == AF_INET6) {
+    inet_addr_.in6_.sin6_port = ACE_HTONS(port);
+#endif
+  }
+}
+
+bool NetworkAddress::is_any() const
+{
+  if (inet_addr_.in4_.sin_family == AF_INET) {
+    return inet_addr_.in4_.sin_addr.s_addr == INADDR_ANY;
+#if defined (ACE_HAS_IPV6)
+  } else if (inet_addr_.in6_.sin6_family == AF_INET6) {
+    return IN6_IS_ADDR_UNSPECIFIED(&inet_addr_.in6_.sin6_addr);
+#endif
+  }
+  return false;
+}
+
+bool NetworkAddress::is_loopback() const
+{
+  if (inet_addr_.in4_.sin_family == AF_INET) {
+    // RFC 3330 defines loopback as any address with 127.x.x.x
+    return (ACE_HTONL(inet_addr_.in4_.sin_addr.s_addr) & 0xFF000000) == (INADDR_LOOPBACK & 0XFF000000);
```

Comment:
Can the right-hand side just be `0XFF000000`?

Reply:
`0x7F000000`, yes.